### ISSUE TITLE
Unable to create Elasticache 6.0 anymore

### ISSUE THREAD

**Issue by obourdon** [hashicorp/terraform-provider-aws#26047]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

AWS provider 3.38.0 -> 3.75.2 and 4.0.0 -> 4.11.0
Any Terrraform version

### Affected Resource(s)

* aws_elasticache

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource "aws_elasticache_replication_group" "redis-cluster" {
  replication_group_id          = "test-redis-cluster"
  replication_group_description = "Redis cluster design to handle events and async work"
  node_type                     = "cache.t3.micro"
  engine_version                = "6.0"
  subnet_group_name             = aws_elasticache_subnet_group.redis-cluster-subnet-group.name
  security_group_ids            = [aws_security_group.redis_sg.id]
  maintenance_window            = "sun:01:00-sun:04:00"
  snapshot_window               = "22:00-00:00"
  snapshot_retention_limit      = 7
  automatic_failover_enabled    = true
  apply_immediately             = true

  cluster_mode {
    replicas_per_node_group = 1
    num_node_groups         = 2
  }
}
```

### Debug Output

N/A

### Panic Output

N/A

### Expected Behavior

Both 6.2 and 6.0 are available in eu-west-1 AWS console and should be also available via AWS provider

### Actual Behavior

Only 6.x can be specified

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

N/A

### References

N/A even though it seems like some other issues are mentioning weird behaviours around the same subject, it
did not seem to me that this specific issue was raised

**Comment by justinretzolk** [comment]
Hey @obourdon 👋 Thank you for taking the time to raise this! So that we have all of the necessary information in order to look into this, can you supply debug logs (redacted as needed) as well?

**Comment by obourdon** [comment]
after re-evaluation it only applies to 3.75.x and below
@justinretzolk AFAICT, there is no need for debug traces as [can be seen here](https://github.com/hashicorp/terraform-provider-aws/blob/v3.75.2/internal/service/elasticache/validation.go#L17)

**Comment by obourdon** [comment]
Going even further, trying back versions one a a time, the issue occurred between 3.37.0 (working) and 3.38.0 (not working)
The latest version inserted a new file as `aws/elasticache_validation.go`

**Comment by obourdon** [comment]
Going one more step further, the same issue occurs in version 4.0.0 till 4.11.0 but is removed in 4.12.0 with the new `engine_version.go` file

**Comment by obourdon** [comment]
@justinretzolk any progress on this ?

**Comment by justinretzolk** [comment]
Hey @obourdon 👋 Thank you for the additional information and updates here. Am I correct in my understanding that with the latest version in the `3.x` series, and in versions after `4.12.0`, this is resolved?

**Comment by obourdon** [comment]
any news on this ?

**Comment by obourdon** [comment]
It does work with latest versions of TF AWS provider (>4.11.0) but it does NOT with 3.75.2 (it has been broken since 3.38)

**Comment by olhado** [comment]
I am seeing this when defining `engine_version = "6.2".

Error is:
```
Error: 1 error occurred:
	* engine_version: Redis versions must match <major>.x when using version 6 or higher, or <major>.<minor>.<bug-fix>
```

We are on the latest 3.X of the aws provider.

**Comment by justinretzolk** [comment]
Hey y'all 👋 Thank you for confirming that this is fixed in AWS provider versions > `4.11.0`. Given that information, we'll close this issue for now, as we are [no longer backporting non-critical fixes](https://hashicorp.github.io/terraform-provider-aws/faq/#once-a-major-release-is-published-will-new-features-and-fixes-be-backported-to-previous-versions) to version `3.x`(with the exception of security vulnerabilities). If you feel we've done this in error, please do let us know and we'd be happy to continue the conversation.
